tests/test_topo_constraints.py

```diff
@@ -0,0 +1,222 @@
+                        cost_ordering_var,
+                        0.0,
+                        err_msg=f"expected the cost order var for {p} and {pc=} to be 0.0, since"
+                        f"{chosen_pc=} with higher investment costs",
+                    )
+                elif pc.investment_costs > chosen_pc.investment_costs:
+                    np.testing.assert_almost_equal(
+                        cost_ordering_var,
+                        1.0,
+                        err_msg=f"Expected the cost order var for {p} and {pc=} to be 1.0, since"
+                        f"{chosen_pc=} with lower investment costs",
+                    )
+
+                discharge_ordering_var_name = base_name + "_discharge_ordering"
+                discharge_ordering_var = self.results[discharge_ordering_var_name]
+                if pc.maximum_discharge < chosen_pc.maximum_discharge:
+                    np.testing.assert_almost_equal(
+                        discharge_ordering_var,
+                        0.0,
+                        err_msg=f"expected the discharge order var for {p} and {pc=} to be 0.0, "
+                        f"since {chosen_pc=} with higher max discharge",
+                    )
+                elif pc.maximum_discharge > chosen_pc.maximum_discharge:
+                    np.testing.assert_almost_equal(
+                        discharge_ordering_var,
+                        1.0,
+                        err_msg=f"Expected the discharge order var for {p} and {pc=} to be 1.0, "
+                        f"since {chosen_pc=} with lower max discharge",
+                    )
+
+                heat_loss_ordering_var_name = base_name + "_heat_loss_ordering"
+                heat_loss_ordering_var = self.results[heat_loss_ordering_var_name]
+                pc_heat_loss = self.get_heat_losses(p, pc)
+                chosen_pc_heat_loss = self.get_heat_losses(p, chosen_pc)
+                if pc_heat_loss < chosen_pc_heat_loss:
+                    np.testing.assert_almost_equal(
+                        heat_loss_ordering_var,
+                        0.0,
+                        err_msg=f"expected the heat loss order var for {p} and {pc=} to be 0.0, "
+                        f"since {chosen_pc=} with higher heat losses",
+                    )
+                elif pc_heat_loss > chosen_pc_heat_loss:
+                    np.testing.assert_almost_equal(
+                        discharge_ordering_var,
+                        1.0,
+                        err_msg=f"Expected the heat loss order var for {p} and {pc=} to be 1.0, "
+                        f"since {chosen_pc=} with lower heat losses",
+                    )
+
+        for pc_name, total_count in pc_sums.items():
+            count_var_value = self.results[f"{pc_name}__global_pipe_class_count"]
+            np.testing.assert_almost_equal(
+                count_var_value,
+                total_count,
+                6,
```

Comment:
I don't get the 6 here, this now is tested if accurate up to the 6th decimal. Why not just use the default settings?

Reply:
I don't think there's a reason. I probably copied the 6 over from the test_pipe_class_var. I'll see if I the default suffices there too.